src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -1670,18 +1670,21 @@ private static Type[] ResolveGenericTypeParameters(object[] genericTypeParameter
 
         internal static Type EffectiveArgumentType(object arg)
         {
-            if (arg != null)
+            arg = PSObject.Base(arg);
+            if (arg is null)
             {
-                arg = PSObject.Base(arg);
-                object[] argAsArray = arg as object[];
-                if (argAsArray != null && argAsArray.Length > 0 && PSObject.Base(argAsArray[0]) != null)
+                return typeof(LanguagePrimitives.Null);
+            }
+
+            if (arg is object[] array && array.Length > 0)
```

Comment:
I'd rather keep the old descriptive name `argAsArray` the new `array` leaves no clue about what the variable holds, its like a `String string` or `int i` - fine in some context, don't see a point in changing it here.

Reply:
The `is` expression makes the variable quite descriptive, as in "arg is an array".